moondream/torch/vision.py

```diff
@@ -19,71 +24,135 @@ def im_resize(
     interpolation: InterpolationMode = InterpolationMode.BICUBIC,
 ) -> Image.Image:
     """
-    The 'resize' function from torchvision has bad type signatures.
-    it accepts both PIL images and torch tensors,  but the type signature
-    only allows tensors.
+    Resize a PIL image using torchvision's resize function.
     """
     return tv_resize(
         image,  # type: ignore
         size,
-        InterpolationMode.BICUBIC,
+        interpolation,
```

Comment:
We want to have this implementation used in the Python client too, which doesn't have a dependency on torch/torchvision. They're pretty heavy dependencies. We should stick with PIL and numpy here, and avoid using torch.